app/src/features/rules/context/actions.tsx

```diff
@@ -242,6 +244,50 @@ export const RulesActionContextProvider: React.FC<RulesProviderProps> = ({ child
     [updateRecordInStorage, userAttributes.num_active_rules]
   );
 
+  const recordsStatusUpdateAction = useCallback(
+    (records: StorageRecord[], value: RecordStatus, onSuccess?: Function) => {
+      Logger.log("[DEBUG]", "recordsStatusUpdateAction", records);
+
+      const handleRecordStatusUpdate = (record: StorageRecord, value: RecordStatus) => {
+        const newStatus = value;
+        const updatedRecord: StorageRecord = {
+          ...record,
+          status: newStatus,
+        };
+
+        Logger.log("Writing storage in RulesTable changeRuleStatus");
```

Comment:
Refractor `recordsStatusToggleAction` to use this method only `recordsStatusUpdateAction`.

Also since `recordsStatusToggleAction` is a action that needs to be performed on a single rule, rename it to `recordStatusToggleAction` and use `recordsStatusUpdateAction` internally